crates/starknet-types-core/src/curve/projective_point.rs

```diff
@@ -0,0 +1,213 @@
+use crate::curve::affine_point::AffinePoint;
+use crate::curve::curve_errors::CurveError;
+use crate::felt::Felt;
+use core::ops;
+use lambdaworks_math::cyclic_group::IsGroup;
+use lambdaworks_math::elliptic_curve::short_weierstrass::curves::stark_curve::StarkCurve;
+use lambdaworks_math::elliptic_curve::short_weierstrass::point::ShortWeierstrassProjectivePoint;
+use lambdaworks_math::elliptic_curve::traits::EllipticCurveError::InvalidPoint;
+use lambdaworks_math::unsigned_integer::traits::IsUnsignedInteger;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ProjectivePoint(pub(crate) ShortWeierstrassProjectivePoint<StarkCurve>);
+
+impl ProjectivePoint {
+    pub fn new(x: Felt, y: Felt, z: Felt) -> ProjectivePoint {
+        Self(ShortWeierstrassProjectivePoint::new([x.0, y.0, z.0]))
+    }
+
+    /// The point at infinity.
+    pub fn identity() -> ProjectivePoint {
+        Self(ShortWeierstrassProjectivePoint::neutral_element())
+    }
+
+    /// Creates the same point in affine coordinates. That is,
+    /// returns [x / z: y / z: 1] where `self` is [x: y: z].
+    /// Panics if `self` is the point at infinity.
```

Comment:
small detail: The doc says it panics but a `Result` is returned

Reply:
Nice Catch!